workflow/controller/dag.go

```diff
@@ -651,22 +651,42 @@ func (woc *wfOperationCtx) resolveDependencyReferences(dagCtx *dagContext, task
 // findLeafTaskNames finds the names of all tasks whom no other nodes depend on.
 // This list of tasks is used as the the default list of targets when dag.targets is omitted.
 func (d *dagContext) findLeafTaskNames(tasks []wfv1.DAGTask) []string {
+
+	prioritySort := false
+
+	// Sort tasks by priority
+	sort.Slice(tasks, func(i, j int) bool {
```

Comment:
With Alex's comment, maybe we should put these into global util?
https://go.dev/doc/tutorial/generics

Reply:
Will put the sort function into a generic so that it works with both tasks and steps.

Reply:
```
func SortByPriority[T []wfv1.WorkflowStep | []wfv1.DAGTask](m T) {
	sort.Slice(m, func(i, j int) bool {

		if m[i].GetPriority() > m[j].GetPriority() {
			return true
		}

		return strings.Compare(m[i].Name, m[j].Name) > 0
	})
}
```

@tczhao This is the generic function I am trying to use to sort both steps and tasks however I keep getting an invalid operation syntax error when trying to index m. 

`cannot index m (variable of type T constrained by []"github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1".WorkflowStep|[]"github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1".DAGTask)`

Reply:
nice